### ISSUE TITLE
[Surface Laptop] Unable to Suspend / Hibernate

### ISSUE THREAD

**Issue by 33Fraise33** [jakeday/linux-surface#409]
Hello,

I'm using this kernel on arch linux from this source: https://github.com/dmhacker/arch-linux-surface
Everytime I try the suspend option: `sudo systemctl suspend` or the hibernate option: `sudo systemctl hibernate` my system shuts down. When I reboot I get the logs 'recovering journal' during boot.

What messages/log should I look into to see what's going wrong? 

During boot I also always get: "nvme nvme0 failed to set APST feature". It does ignore this though.

**Comment by tkimmm** [comment]
Whats your setup (bootloader, OS's etc)

**Comment by 33Fraise33** [comment]
> Whats your setup (bootloader, OS's etc)

I'm running Grub 2.02 with Arch as stated above.

**Comment by tkimmm** [comment]
I would start by looking at whether your swap partition/file is properly configured, I'm running arch too with no problems with suspend, albeit I opted not to replace suspend/hibernate during the setup script.

**Comment by StollD** [comment]
Did you configure GRUB / kernel / initramfs to resume from the swap partition? For me (Arch as well, SB2) it worked perfectly after following these steps from the [Arch Wiki](https://wiki.archlinux.org/index.php/Power_management/Suspend_and_hibernate#Hibernation). 

Especially make sure you also add the initramfs module that is described at the bottom, since leaving that out broke the resuming for me and it cold booted.

**Comment by sphh** [comment]
Just a stupid question: Is your swap partition encrypted? That's what keeps me from hibernating.

**Comment by 33Fraise33** [comment]
> Did you configure GRUB / kernel / initramfs to resume from the swap partition? For me (Arch as well, SB2) it worked perfectly after following these steps from the [Arch Wiki](https://wiki.archlinux.org/index.php/Power_management/Suspend_and_hibernate#Hibernation).
> 
> Especially make sure you also add the initramfs module that is described at the bottom, since leaving that out broke the resuming for me and it cold booted.

This is interesting I will try that and update once I tested this. 



> Just a stupid question: Is your swap partition encrypted? That's what keeps me from hibernating.

no it is not

**Comment by 33Fraise33** [comment]
I went through setting my kernel parameter: `resume=/dev/nvme0n1p5 swap_file_offset=419840` (offset because of swapfile)
I also set my hooks in mkinitcpio.conf: `HOOKS=(base udev autodetect modconf block filesystems keyboard resume fsck)` Afterwards I installed the latest kernel again to trigger the built of initramfs. I created my grubconfig and I still get the same error.

Running from a different TTY I get the following error: 
```
ipts mei::<id>: error in reading m2h msg
nvme nvme0: failed to set APST feature: (24578)
```

**Comment by StollD** [comment]
> I went through setting my kernel parameter: `resume=/dev/nvme0n1p5 swap_file_offset=419840` (offset because of swapfile)

The kernel parameter needs to be `resume_offset`, not `swap_file_offset`. Sadly the Arch Wiki is very misleading on that point, since it only uses the actual parameter name once directly under the big btrfs warning banner), and then constantly refers to `swap_file_offset` which is just a fill-in for the actual value.

**Comment by 33Fraise33** [comment]
> > I went through setting my kernel parameter: `resume=/dev/nvme0n1p5 swap_file_offset=419840` (offset because of swapfile)
> 
> The kernel parameter needs to be `resume_offset`, not `swap_file_offset`. Sadly the Arch Wiki is very misleading on that point, since it only uses the actual parameter name once directly under the big btrfs warning banner), and then constantly refers to `swap_file_offset` which is just a fill-in for the actual value.

I'm now able to hibernate my computer. After reboot the machine completely hangs and I have to reboot. 
I have no idea how to continue from this.

In my TTY I see: 
```
[ 1026.429691] [drm:intel_guc_send_mmio [i915]] *ERROR* MMIO: GuC action 0x10 failed with error -110 0x10
[ 1026.429712] [drm:i915_guc_ipts_reacquire_doorbell [i915]] *ERROR* Not able to reacquire IPTS doorbell
```

Same as issue: https://github.com/jakeday/linux-surface/issues/395

**Comment by qzed** [comment]
Yeah, the last part looks like #395, let's track that over there.
